src/components/DatePicker/DatePicker.tsx

```diff
@@ -299,32 +277,6 @@ function handleKeyDown(event: React.KeyboardEvent<HTMLElement>) {
   }
 }
 
-function isSameSelectedDate(
```

Comment:
Are we sure we can remove this logic?

Reply:
Yeah, I'm not even sure why we're doing this.

This is just doing an actual value comparison, instead of just trusting that if the reference is different then something has changed. I think whoever wrote this was forced to do it like this to avoid re-rendering unnecessarily, but now I can just replace this with the `useEffect` hook in line 69 that only fires if the prop actually changed.

The only case I can think that this doesn't account for is someone selecting the same date twice in a row, and triggering a re-render of the component. But I tried that and it seemed to work fine, so

Reply:
Looking back the code seems to fix [this issue](https://github.com/Shopify/polaris-react-deprecated/pull/1399) and a quick 🎩 shows it's still 💯 ✅ 

> useEffect hook in line 69 that only fires if the prop actually changed

Funny enough we could have done this in a class based component as well.

```tsx
  useEffect(() => {
    setFocusDate(undefined);
  }, [selected]);
```
Is essentially the same as

```tsx
componentDidUpdate(prevProps) {
 if (!Object.is(prevProps.selected, this.props.selected) 
    this.setState({focusedDate: undefined})
}
```